python/tskit/tables.py

```diff
@@ -3020,6 +3026,99 @@ def union(
                 record=json.dumps(provenance.get_provenance_dict(parameters))
             )
 
+    def time_join(self, recent, time, continue_nodes=None):
```

Comment:
Probably more intuitive to call this parameter ``join_nodes``, or perhaps just ``nodes``?

Also, I think we can make ``time`` an optional parameter, making it default to the time of the oldest root in ``recent_ts``